### DISCUSSION TITLE
Import Obsidian Vault

### DISCUSSION THREAD

**Discussion by geodose** [outline/outline#4601] [Ideas]
Adding the ability to import Obsidian Vaults would be very useful. This may be currently possible through the API, but having the functionality alongside the current Notion and .md import options would be great!

**Comment by ReessKennedy** [comment]
I am trying to find some way to have Outline and Obsidian work together. The best hope right now would be uni-directional sync instead of bi-directional. This would mean:
- using the API to auto export docs to a git repository that I keep inside my Obsidian vault and can update
- I can then make a branch of this for my working updates in between syncs and then head to the web editor when I want to update my team notes with stuff I have written locally and then

Any thoughts on this appreciated! I haven't cracked into it yet as I was exploring other alternatives like maybe just using Github's wikis, which would be simpler--or spinning up Gollum--but the web editor and features of Outline are hard to overlook and writing some bridge that would allow for local editing may end up being worthwhile.

**Comment by YendisFish** [comment]
Something like this would actually be extremely useful. Even just some sort of support for Obsidian markdown would be a good start.

**Reply by tommoor** [reply]
This isn't going to be a priority for the maintaining team, but feel free to build something! This is why we have a full open API

**Comment by defcon1702** [comment]
I created today a small plugin that allows you to push folders / notes one way to Outline.
Maybe you want to take a look?

https://github.com/defcon1702/obsidian-outline

Friendly warning: I used Windsurf together with Claude 4.6 to create this so please try it out with caution. I tested it with several cases and seems to work out pretty well. I also made a security audit and fixed everything critical. BUT: please be safe and try it out with a small test vault / test notes.

It should be able to: 
- push folders (top folder won't get a parent note and will be pushed in your Root, Subfolders will be handled as subnotes)
- duplicate detection and handling
- you can setup a standard folder or leave this setting empty -> a modal will show up every push and ask for the folder you want the note in

Known issues / not tested:
- duplicate note detection is over all folders. I don't see this as a real issue, your note only get a -number suffix.

Let me know what you think. :)

**Reply by ReessKennedy** [reply]
Really well done! But just to be clear this this solves the share problem in that you can now share all your notes from Obsidian easily on outline, but if you're using outline for collaboration, someone makes an outline then that will be overwritten if you push again from your local version because it's not aware of the remote changes, right?

**Reply by defcon1702** [reply]
It's not a 2-way sync (not an easy task and often not needed) but it has a duplication detection. It detects a note with the same name and ask if you want to duplicate or overwrite. This worked for me.
But as always: be safe and test it with some random notes. :)
